pkg/reconciler/isbsvc/installer/jetstream.go

```diff
@@ -247,17 +249,17 @@ func (r *jetStreamInstaller) createStatefulSet(ctx context.Context) error {
 func (r *jetStreamInstaller) createSecrets(ctx context.Context) error {
 	oldServerObjExisting, oldClientObjExisting := true, true
 
-	oldSObj := &corev1.Secret{}
-	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.isbs.Namespace, Name: generateJetStreamServerSecretName(r.isbs)}, oldSObj); err != nil {
+	oldSObj, err := r.kubeClient.CoreV1().Secrets(r.isbs.Namespace).Get(ctx, generateJetStreamServerSecretName(r.isbs), metav1.GetOptions{})
+	if err != nil {
 		if apierrors.IsNotFound(err) {
 			oldServerObjExisting = false
 		} else {
 			return fmt.Errorf("failed to check if nats server auth secret is existing, err: %w", err)
 		}
 	}
 
-	oldCObj := &corev1.Secret{}
-	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.isbs.Namespace, Name: generateJetStreamClientAuthSecretName(r.isbs)}, oldCObj); err != nil {
+	oldCObj, err := r.kubeClient.CoreV1().Secrets(r.isbs.Namespace).Get(ctx, generateJetStreamClientAuthSecretName(r.isbs), metav1.GetOptions{})
```

Comment:
Controller-runtime client `Get()` needs `watch` privilege, change to use k8s client.